pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -253,20 +309,47 @@ func (tc *testCase) verifyResults(t *testing.T) {
 	}
 }
 
-func (tc *testCase) runTest(t *testing.T) {
-	testClient := tc.prepareTestClient(t)
-	metricsClient := metrics.NewHeapsterMetricsClient(testClient, metrics.DefaultHeapsterNamespace, metrics.DefaultHeapsterScheme, metrics.DefaultHeapsterService, metrics.DefaultHeapsterPort)
-	hpaController := NewHorizontalController(testClient.Core(), testClient.Extensions(), testClient.Extensions(), metricsClient, 0)
-	stop := make(chan struct{})
-	defer close(stop)
-	go hpaController.Run(stop)
-	if tc.verifyEvents {
-		// We need to wait for events to be broadcasted (sleep for longer than record.sleepDuration).
-		time.Sleep(12 * time.Second)
+func runTest(t *testing.T, tc *testCase) {
```

Comment:
This function makes copies of `tc`, one for each kind. It looks very strange to make copies of the method receiver and then call further methods on the copies. So I made this change.